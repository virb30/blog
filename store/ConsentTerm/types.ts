import { MutationTree, ActionContext, ActionTree, GetterTree } from 'vuex';

export interface State {
	cookiesPolicyAccepted: Boolean;
}

export type RootState = ReturnType<() => State>

export interface Getters extends GetterTree<RootState, State> {
	getPolicyAccepted: (state: State) => Boolean;
}

export enum Mutations {
	SET_COOKIES_ACCEPTED = 'SET_COOKIES_ACCEPTED',
}

export interface MutationsInterface extends MutationTree<RootState> {
	[Mutations.SET_COOKIES_ACCEPTED](s:State, p: Boolean): void;
}

export enum Actions {
	acceptCookiesPolicy = 'acceptCookiesPolicy',
	fetchFromStorage = 'fetchFromStorage'
}

export interface ActionsInterface extends ActionTree<RootState, RootState> {
	acceptCookiesPolicy(actionContext: ActionContext<RootState, RootState>): void
	fetchFromStorage(actionsContext: ActionContext<RootState, RootState>): void
}
