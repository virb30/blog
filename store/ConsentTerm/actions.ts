import { ActionsInterface, Mutations } from './types';

export default {
	acceptCookiesPolicy ({ commit }) {
		sessionStorage.setItem('cookie-policy-accepted', JSON.stringify(true));
		commit(Mutations.SET_COOKIES_ACCEPTED, true);
	},
	fetchFromStorage ({ commit }) {
		const accepted = !!sessionStorage.getItem('cookie-policy-accepted');
		commit(Mutations.SET_COOKIES_ACCEPTED, accepted);
	},
} as ActionsInterface;
