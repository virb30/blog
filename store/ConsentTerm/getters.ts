import { RootState, State } from './types';
import { GetterTree } from 'vuex';

const getters: GetterTree<RootState, State> = {
	getPolicyAccepted: (state) => {
		return state.cookiesPolicyAccepted
	},
};

export default getters;
