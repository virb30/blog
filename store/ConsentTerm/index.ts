import { State } from './types';

export const state = (): State => ({
	cookiesPolicyAccepted: false,
});
