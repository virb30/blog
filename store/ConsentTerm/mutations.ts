
import { Mutations, MutationsInterface } from './types';

export default {
	[Mutations.SET_COOKIES_ACCEPTED] (state, accepted) {
		state.cookiesPolicyAccepted = accepted;
	},
} as MutationsInterface;
