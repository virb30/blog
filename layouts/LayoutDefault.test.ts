import Vuex from 'vuex';
import VueMeta from "vue-meta";
import { createLocalVue, mount } from "@vue/test-utils";
import Layout from './default.vue';
import ConsentTerm from '../components/CookieConsent/index.vue';

const localVue = createLocalVue();
localVue.use(VueMeta, { keyName: 'head' });
localVue.use(Vuex);


const buildConsentTermStore = (config: any = {}) => {
	const mockGetter = jest.fn();
	mockGetter.mockImplementationOnce(() => false);

	const mockFetchFromStorage = jest.fn();
	mockFetchFromStorage.mockImplementationOnce(() => false);

	const defaultConfig = {
		namespaced: true,
		state: {
			cookiesPolicyAccepted: false
		},
		getters: {
			getPolicyAccepted: mockGetter,
		},
		actions: {
			fetchFromStorage: mockFetchFromStorage,
		}
	};

	return new Vuex.Store({
		modules: {
			ConsentTerm: {
				...defaultConfig,
				...config
			}
		}
	});
}

describe('Default Layout', () => {
	let store: any;
	let config: any;

	beforeEach(() => {
		store = buildConsentTermStore();
		config = {
			localVue,
			mocks: {
				$route: {
					path: ''
				},
				$colorMode: {
					preference: 'dark',
				}
			},
			store
		}
	})

	it('should render childs', () => {
		const wrapper = mount(Layout, config);

		const div = wrapper.find('div');

		expect(wrapper.html()).toMatchSnapshot();
		expect(div.classes()).toContain('dark');
	});

	it('should render cookies consent term dialog', () => {
		const wrapper = mount(Layout, config);

		const cookieConsentTerm = wrapper.findComponent(ConsentTerm);

		expect(wrapper.html()).toMatchSnapshot();
		expect(cookieConsentTerm.exists()).toBe(true);
	});

	it('should not render cookies consent term dialog', () => {
		const mockGetter = jest.fn();
		mockGetter.mockImplementationOnce(() => true);

		store = buildConsentTermStore({
			getters: {
				getPolicyAccepted: mockGetter,
			},
		});

		const wrapper = mount(Layout, {...config, store});

		const cookieConsentTerm = wrapper.findComponent(ConsentTerm);

		expect(wrapper.html()).toMatchSnapshot();
		expect(cookieConsentTerm.exists()).toBe(false);
	});
});
