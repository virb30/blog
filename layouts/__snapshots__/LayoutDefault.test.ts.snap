// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Layout should not render cookies consent term dialog 1`] = `
"<div class=\\"w-full dark\\">
  <header class=\\"h-20 border-0 border-b border-solid border-gray-400 dark:border-gray-100\\">
    <div class=\\"max-w-7xl h-20 my-0 mx-auto p-4 flex items-center justify-between\\">
      <logo class=\\"max-w-xs hidden md:block lg:block xl:block\\"></logo>
      <nav class=\\"md:ml-20 h-20\\">
        <nuxtlink exact=\\"\\" to=\\"/\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Home
        </nuxtlink>
        <nuxtlink to=\\"/about\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Sobre
        </nuxtlink>
        <nuxtlink to=\\"/blog\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Blog
        </nuxtlink>
      </nav>
      <div class=\\"ml-auto p-4\\">
        <switchmode></switchmode>
      </div>
    </div>
  </header>
  <nuxt></nuxt>
  <!---->
</div>"
`;

exports[`Default Layout should render childs 1`] = `
"<div class=\\"w-full dark\\">
  <header class=\\"h-20 border-0 border-b border-solid border-gray-400 dark:border-gray-100\\">
    <div class=\\"max-w-7xl h-20 my-0 mx-auto p-4 flex items-center justify-between\\">
      <logo class=\\"max-w-xs hidden md:block lg:block xl:block\\"></logo>
      <nav class=\\"md:ml-20 h-20\\">
        <nuxtlink exact=\\"\\" to=\\"/\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Home
        </nuxtlink>
        <nuxtlink to=\\"/about\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Sobre
        </nuxtlink>
        <nuxtlink to=\\"/blog\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Blog
        </nuxtlink>
      </nav>
      <div class=\\"ml-auto p-4\\">
        <switchmode></switchmode>
      </div>
    </div>
  </header>
  <nuxt></nuxt>
  <div id=\\"consent-term-container\\" class=\\"bg-gray-900 text-gray-50 dark:bg-gray-50 dark:text-gray-900 fixed bottom-0 w-full\\">
    <div class=\\"py-6 px-8 flex items-center justify-center max-w-6xl m-auto\\">
      <p class=\\"text-sm\\">
        Nós usamos Google Analytics para realizar análises estatísticas de uso da página e interações para avaliar e desenvolver nosso site. Ao utilizar nossos serviços você concorda com a coleta dessas informações.
        Para saber mais sobre como o Google utiliza os dados coletados através desse serviço, acesse: <a href=\\"http://www.google.com/policies/privacy/partners/\\" class=\\"underline\\">http://www.google.com/policies/privacy/partners/</a></p> <button type=\\"button\\" class=\\"ml-4 border-2 border-gray-50 rounded px-4 h-8 text-xs text-gray-50 hover:bg-gray-50 hover:text-gray-900 dark:border-gray-900 dark:text-gray-900 dark-hover:bg-gray-900 dark-hover:text-gray-50\\">
        Aceitar
      </button>
    </div>
  </div>
</div>"
`;

exports[`Default Layout should render cookies consent term dialog 1`] = `
"<div class=\\"w-full dark\\">
  <header class=\\"h-20 border-0 border-b border-solid border-gray-400 dark:border-gray-100\\">
    <div class=\\"max-w-7xl h-20 my-0 mx-auto p-4 flex items-center justify-between\\">
      <logo class=\\"max-w-xs hidden md:block lg:block xl:block\\"></logo>
      <nav class=\\"md:ml-20 h-20\\">
        <nuxtlink exact=\\"\\" to=\\"/\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Home
        </nuxtlink>
        <nuxtlink to=\\"/about\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Sobre
        </nuxtlink>
        <nuxtlink to=\\"/blog\\" active-class=\\"active\\" class=\\"hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\">
          Blog
        </nuxtlink>
      </nav>
      <div class=\\"ml-auto p-4\\">
        <switchmode></switchmode>
      </div>
    </div>
  </header>
  <nuxt></nuxt>
  <div id=\\"consent-term-container\\" class=\\"bg-gray-900 text-gray-50 dark:bg-gray-50 dark:text-gray-900 fixed bottom-0 w-full\\">
    <div class=\\"py-6 px-8 flex items-center justify-center max-w-6xl m-auto\\">
      <p class=\\"text-sm\\">
        Nós usamos Google Analytics para realizar análises estatísticas de uso da página e interações para avaliar e desenvolver nosso site. Ao utilizar nossos serviços você concorda com a coleta dessas informações.
        Para saber mais sobre como o Google utiliza os dados coletados através desse serviço, acesse: <a href=\\"http://www.google.com/policies/privacy/partners/\\" class=\\"underline\\">http://www.google.com/policies/privacy/partners/</a></p> <button type=\\"button\\" class=\\"ml-4 border-2 border-gray-50 rounded px-4 h-8 text-xs text-gray-50 hover:bg-gray-50 hover:text-gray-900 dark:border-gray-900 dark:text-gray-900 dark-hover:bg-gray-900 dark-hover:text-gray-50\\">
        Aceitar
      </button>
    </div>
  </div>
</div>"
`;
